conda_smithy/lint_recipe.py

```diff
@@ -1105,17 +1108,47 @@ def _format_validation_msg(error: "jsonschema.ValidationError"):
     )
 
 
+def _format_rattler_validation_msg(error: "jsonschema.ValidationError"):
+    return cleandoc(
+        f"""
+        In recipe.yaml: `{error.instance}`.
+{indent(error.message, " " * 12 + "> ")}
+        """
+    )
+
+
+def _lint_recipe_yaml(recipe_path):
+    with open(recipe_path) as fh:
+        meta = get_yaml().load(fh)
+
+    return validate_json_schema(
+        meta, schema_file=RATTLER_RECIPE_YAML_SCHEMA_FILE
+    )
```

Comment:
The linter is supposed to enforce best practices. For eg: checking that the maintainers exist, tests are there, etc.
All this does is that the recipe follows the standard. This is less critical as rattler itself would be doing that.

Reply:
done - implemented linting checks and extended test-suite for this